### PR TITLE
Allow setting/unsetting BYPASSRLS Postgres role attribute

### PR DESCRIPTION
**Pull request by herrewig**
##### SUMMARY
Add the `BYPASSRLS` Postgres role attribute into the list of valid role attributes, allowing for the setting and unsetting of `BYPASSRLS`.

Row security is a feature introduced in Postgres 9.5 that allows for "policies that restrict, on a per-user basis, which rows can be returned by normal queries or inserted, updated, or deleted by data modification commands." There are instances where it would be beneficial for certain roles to bypass row security by setting the `BYPASSRLS` role attribute (or removing the attribute  with `NOBYPASSRLS`).

This simple change allows for the setting or unsetting of the "bypass row security" attribute using `BYPASSRLS` or `NOBYPASSRLS`.

See [the official Postgres Row Security docs](https://www.postgresql.org/docs/9.5/static/ddl-rowsecurity.html) for more details on row security and the BYPASSRLS attribute.

##### ISSUE TYPE
 - Bug Fix

##### COMPONENT NAME
postgresql_user module

##### ANSIBLE VERSION

```
2.2.0.0
```

#### Output
*Before (with high verbosity)*
```
fatal: [myhost]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_args": {
            "db": "", 
            "encrypted": false, 
            "expires": null, 
            "fail_on_user": true, 
            "login_host": "", 
            "login_password": "", 
            "login_unix_socket": "", 
            "login_user": "postgres", 
            "name": "<redacted>", 
            "no_password_changes": false, 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "port": "5432", 
            "priv": null, 
            "role_attr_flags": "BYPASSRLS", 
            "state": "present", 
            "user": "<redacted>"
        }, 
        "module_name": "postgresql_user"
    }, 
    "msg": "Invalid role_attr_flags specified: BYPASSRLS"
}
```

*After (with high verbosity)*
```
changed: [myhost] => {
    "changed": true, 
    "invocation": {
        "module_args": {
            "db": "", 
            "encrypted": false, 
            "expires": null, 
            "fail_on_user": true, 
            "login_host": "", 
            "login_password": "", 
            "login_unix_socket": "", 
            "login_user": "postgres", 
            "name": "<redacted>", 
            "no_password_changes": false, 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "port": "5432", 
            "priv": null, 
            "role_attr_flags": "BYPASSRLS", 
            "state": "present", 
            "user": "<redacted>"
        }, 
        "module_name": "postgresql_user"
    }, 
    "user": "<redacted>"
}
```

### CODE REVIEW

**Comment by nerzhul** [comment]
Hello,
thanks for your addition. Please add a postgresql version check before any application of this new parameter to ensure this will work with postgresql < 9.5

**Comment by herrewig** [comment]
Good idea, @nerzhul -- I believe the latest commit will now address your concern. Kindly let me know what you think :)

I wasn't sure if I should wrap the [version getting and comparison](https://github.com/herrewig/ansible/blob/bb469b7219bc4d5b9cacc376a63e742d51494ed5/lib/ansible/modules/database/postgresql/postgresql_user.py#L664-L670 ) in a try/except to defend against odd versions, but I've run the `SHOW SERVER_VERSION` across multiple Postgres versions, and it appears to always returns a valid semver version , so I think it should be OK the way it is.

**Comment by nerzhul** [comment]
it's okay for me

shipit

**Comment by herrewig** [comment]
Awesome! Thanks @nerzhul . What are the next steps here for getting it merged? Is there anything I need to do?

**Comment by nerzhul** [comment]
you can ask more review on ansible-devel on freenode permitting this to be merged for 2.4

**Comment by herrewig** [comment]
@nerzhul I added some integration tests per the suggestion of freenode chat

*tl;dr* - The integration test will now get the version of Postgres and then import the correct task playbook for testing role attribute management :)

File: test/integration/targets/postgresql/tasks/main.yml [hunk 1]
```diff
@@ -268,92 +268,23 @@
...
-  postgresql_user:
-    name: "{{ db_user1 }}"
-    state: "present"
-    role_attr_flags: "LOGIN"
-    login_user: "{{ pg_user }}"
-    db: postgres
-  register: result
+- name: Role attribute testing for Postgres 9.5+
+  include: postgresql_user_9.5_or_greater.yml
+  when: "{{ (postgres_version_resp.stdout_lines[-2] | trim) | version_compare('9.5.0', '>=') }}"
```
Comment (pilou-) [inline test/integration/targets/postgresql/tasks/main.yml]: Could you fix these warnings ? All Jinja2 delimiters should be removed from `when` statements.
```
2017-05-24 20:23:44  [WARNING]: when statements should not include jinja2 templating delimiters
2017-05-24 20:23:44 such as {{ }} or {% %}. Found: {{ (postgres_version_resp.stdout_lines[-2] |
2017-05-24 20:23:44 trim) | version_compare('9.5.0', '>=') }}
```

**Review by pilou- — COMMENTED** [review]
Thanks for updating the tests ! I checked that both cases are currently tested by the CI (9.3.16 is used by Docker image ubuntu1404, 9.5.6 is used by Docker image ubuntu1604.

File: test/integration/targets/postgresql/tasks/main.yml [hunk 2]
```diff
@@ -268,92 +268,23 @@
     that:
       - "result.changed == False"
 
-- name: Create a user with all role attributes
+# BYPASSRLS role attribute was introduced in Postgres 9.5, so 
+# we want to test atrribute management differently depending
+# on the version. See https://github.com/ansible/ansible/pull/24625
+# for more details.
+- name: Get Postgres version
```
Comment (pilou-) [inline test/integration/targets/postgresql/tasks/main.yml:275]: You could add a debug task here in order to print the PostgreSQL version tested. The version appears only when verbose is enabled.

**Comment by herrewig** [comment]
shipit

**Comment by pilou-** [comment]
shipit

**Comment by pilou-** [comment]
botstatus

**Comment by jctanner** [comment]
@pilou- bot_status

**Comment by nerzhul** [comment]
shipit @ansibot

**Comment by bcoca** [comment]
quick review, LGTM, if no one objects I'll merge tomorrow

**Comment by pilou-** [comment]
bot_status

**Comment by pilou-** [comment]
@jctanner besides only core team, submitter or module maintainers are allowed to use the `bot_status` command :-/ Should we change that ?

**Comment by matburt** [comment]
shipit

**Comment by nerzhul** [comment]
bot_status
